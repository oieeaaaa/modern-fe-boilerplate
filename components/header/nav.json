[
  {
    "label": "🏠 Home",
    "path": "/"
  },
  {
    "label": "🔥 Trending",
    "path": "/trending"
  },
  {
    "label": "📈 Popular",
    "path": "/popular"
  },
  {
    "label": "⚙️ Settings",
    "path": "/settings"
  }
]
