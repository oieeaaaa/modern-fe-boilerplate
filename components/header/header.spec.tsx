import render, { screen } from 'test-utils'
import nav from './nav.json'
import Header from '.'

describe('Header', () => {
  let useRouter = jest.spyOn(require('next/router'), 'useRouter')

  it('should pass snapshot', () => {
    useRouter.mockReturnValueOnce({ asPath: '/' })

    render(<Header />)

    const header = screen.getByTestId('header')

    expect(header).toBeInTheDocument()
    expect(header).toMatchSnapshot()
  })

  it('should change the active if on different location', () => {
    useRouter.mockReturnValueOnce({ asPath: nav[1].path })

    render(<Header />)

    const header = screen.getByTestId('header')
    const activeNavItem = header.querySelector('.active')

    expect(activeNavItem).toHaveTextContent(nav[1].label)
  })
})
