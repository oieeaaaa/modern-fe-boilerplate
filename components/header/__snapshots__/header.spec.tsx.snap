// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should pass snapshot 1`] = `
<header
  class="container header"
  data-testid="header"
>
  <div
    class="header_top"
  >
    <p
      class="header_brand"
    >
      Modern FE Boilerplate
    </p>
    <a
      class="header_search"
      href="/search"
    >
      <svg
        aria-hidden="true"
        class="header_icon"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </a>
    <a
      class="header_cart"
      href="/cart"
    >
      <svg
        aria-hidden="true"
        class="header_icon"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15 17h5l-1.405-1.405A2.032 2.032 0 0118 14.158V11a6.002 6.002 0 00-4-5.659V5a2 2 0 10-4 0v.341C7.67 6.165 6 8.388 6 11v3.159c0 .538-.214 1.055-.595 1.436L4 17h5m6 0v1a3 3 0 11-6 0v-1m6 0H9"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </a>
  </div>
  <nav
    class="header_nav"
  >
    <a
      class="header_nav__item active"
      href="/"
    >
      🏠 Home
    </a>
    <a
      class="header_nav__item"
      href="/trending"
    >
      🔥 Trending
    </a>
    <a
      class="header_nav__item"
      href="/popular"
    >
      📈 Popular
    </a>
    <a
      class="header_nav__item"
      href="/settings"
    >
      ⚙️ Settings
    </a>
  </nav>
</header>
`;
